src/main/java/edu/kit/kastel/sdq/artemis4j/grading/git/VCSTokenCloningStrategy.java

```diff
@@ -0,0 +1,55 @@
+/* Licensed under EPL-2.0 2024. */
+package edu.kit.kastel.sdq.artemis4j.grading.git;
+
+import java.time.ZonedDateTime;
+
+import edu.kit.kastel.sdq.artemis4j.ArtemisNetworkException;
+import edu.kit.kastel.sdq.artemis4j.client.UserDTO;
+import edu.kit.kastel.sdq.artemis4j.grading.ArtemisConnection;
+import org.eclipse.jgit.api.CloneCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Cloning strategy that uses a personal access token (PAT) to clone a repository.
+ */
+public class VCSTokenCloningStrategy implements CloningStrategy {
+    private static final Logger log = LoggerFactory.getLogger(VCSTokenCloningStrategy.class);
+
+    private final CredentialsProvider credentialsProvider;
+
+    public VCSTokenCloningStrategy(String tokenOverride, ArtemisConnection connection) throws ArtemisNetworkException {
+        var assessor = connection.getAssessor();
+
+        String token;
+        if (tokenOverride != null) {
+            token = tokenOverride;
+        } else if (assessor.getGitToken().isPresent()) {
+            token = assessor.getGitToken().get();
+        } else {
+            // The user has not set a token, so create one for him with a default expiration time
+            log.info("Creating new VCS access token");
+            var expiryDate = ZonedDateTime.now().plusDays(5);
```

Comment:
Please check how long the max time currently is (hard coded in artemis). I'd create a token at least for some weeks :)

Reply:
The max time is one year (https://github.com/ls1intum/Artemis/blob/921b1884fb1a3028512e61023f254f1365fec14b/src/main/java/de/tum/in/www1/artemis/web/rest/AccountResource.java#L175), but that feels way too long to me. What about 6 months? That ensures that tokens generated at the start of a semester are invalidated at its end.

Reply:
Fine for me :)